src/main/java/com/ably/kafka/connect/utils/RecordHeaderConversions.java

```diff
@@ -0,0 +1,136 @@
+    private static final String PUSH_HEADER = "com.ably.extras.push";
+    private static final String KAFKA_KEY = "kafka";
+    private static final String PUSH_KEY = "push";
+    /**
+     * Returns the extras object to use when converting a Kafka message
+     * to an Ably message.
+     * <p>
+     * If the Kafka message has a key, it is base64 encoded and set as the
+     * "key" field in the extras and is added under "kafka" extras.
+     * <p>
+     * If the Kafka message has headers, they are set as the "headers" field
+     * in the extras.
+     *
+     * If the Kafka message has "com.ably.extras.push" header this is set as "push" extras for Ably message.
+     *
+     * @param record The sink record representing the Kafka message
+     * @return The Kafka message extras object
+     */
+    @Nullable
+    public static MessageExtras toMessageExtras(final SinkRecord record) {
+        JsonUtils.JsonUtilsObject kafkaExtras = null;
+        JsonUtils.JsonUtilsObject extrasObject = null;
+
+        Object pushExtras = null;
+        final byte[] key = (byte[]) record.key();
+        if (key != null) {
+            kafkaExtras = JsonUtils.object();
+            final String keyValue = Base64.getEncoder().encodeToString(key);
+            kafkaExtras.add("key", keyValue);
+        }
+
+        if (record.headers().isEmpty()){
+            if (kafkaExtras == null) {
+                return null;
+            }
+            extrasObject = JsonUtils.object();
+            extrasObject.add(KAFKA_KEY, kafkaExtras);
+            return new MessageExtras(extrasObject.toJson());
+        }
+
+        JsonUtils.JsonUtilsObject headersObject = null;
+        for (Header header : record.headers()) {
+            if (header.key().equals(PUSH_HEADER)) {
+                pushExtras = header.value();
+            } else {
+                if (kafkaExtras == null) {
+                    kafkaExtras = JsonUtils.object();
+                }
+                if (headersObject == null) {
+                    headersObject = JsonUtils.object();
+                }
+                headersObject.add(header.key(), header.value());
+            }
+        }
+
+        if (kafkaExtras == null && pushExtras == null) {
+            return null;
+        }
+
+        if (kafkaExtras != null && headersObject != null) {
+            kafkaExtras.add("headers", headersObject);
+        }
+
+        if (kafkaExtras != null) {
+            extrasObject = JsonUtils.object();
+            extrasObject.add(KAFKA_KEY, kafkaExtras);
+        }
+
+        if (pushExtras != null) {
+            String pushExtrasJson = null;
+            if (pushExtras instanceof String){
+                pushExtrasJson = (String) pushExtras;
+            } else if (pushExtras instanceof Map) {
+                pushExtrasJson = new Gson().toJson(pushExtras, Map.class);
+            }
+            if (pushExtrasJson != null){
```

Comment:
A bit of inconsistent spacing on some of the `if` and other statements - some have a space before `{`,  others dont

Reply:
This will also hopefully have been addressed by 1918c37cf4bf6bb6c23d26869153db9b0ba24eba